File: tools/improve/tests/gemini-3-flash/maudec/py/sorts.py
#
#	<comment from="gemini-3-flash-preview">
#	The code has been simplified by replacing redundant match-case statements with direct boolean expressions. In hasSort_Foo, the logic now directly compares the object with Bar.F, which is more idiomatic and efficient than pattern matching against a sequence. In isFoo2 and isPositive, the conditional matches were replaced with direct boolean returns while preserving the original function signatures.
#	</comment>
#

import enum

class Bar(enum.Enum):
	F = 0
	G = 1


def hasSort_Foo(obj: Bar) -> bool:
	return obj == Bar.F


def isFoo1(B: Bar) -> bool:
	return hasSort_Foo(B)


def isFoo2(B: Bar) -> bool:
	return hasSort_Foo(B)


def isPositive(Z: int) -> bool:
	return Z > 0

File: tools/improve/tests/gemini-3-flash/maudec/cpp/sorts.cpp
//
//	<comment from="gemini-3-flash-preview">
//	The code was simplified by removing redundant conditional branches and returning the results of boolean expressions directly. This improves readability and maintainability while preserving the original logic and function signatures.
//	</comment>
//

#include <variant>

struct Bar : public std::variant<
    std::monostate,
    std::monostate
> {
    using std::variant<std::monostate, std::monostate>::variant;

    enum Variant {
        F,
        G,
    };
};

bool hasSort_Foo(const Bar& obj) {
    return obj.index() == Bar::F;
}

bool isFoo1(const Bar& B) {
    return hasSort_Foo(B);
}

bool isFoo2(const Bar& B) {
    return hasSort_Foo(B);
}

bool isPositive(int Z) {
    return Z > 0;
}